the-collab-lab/tcl-79-smart-shopping-list
# Feat: Delete lists

## Description

Users can now delete lists from the home page. They are only allowed to delete lists they have created themselves, the button is disabled for lists that have been shared with the logged user.
Deleting a list removes the corresponding document in the user's collecton, and the reference to it in the sharedLists field array.

## Related Issue

Closes #36 

## Acceptance Criteria

- [x] The delete button is active for user created lists, and inactive for shared ones.
- [x] When clicked, the delete button immediately removes the list from Your Lists.
- [x] After a list is deleted, it disappears from the user collection and from the sharedLists field array. 

## Type of Changes

`feat` 

## Updates

### Before

Pretty much the same.

### After
Disabled button is now grey.
![image](https://github.com/user-attachments/assets/fdf70494-d0b5-404b-97dc-63c6daf6f69a)

No list selected
![image](https://github.com/user-attachments/assets/009f0c2b-18fc-4512-b6be-9aa36c306afb)


## Testing Steps / QA Criteria

- Do a `git pull` and then `git checkout qg-feat-delete-list`
- Run `npm start` to open GrocerEase
- Create a few test lists, then click their delete buttons to delete them.
- After deletion, the lists should be gone from both your user collection and your sharedLists array.
- If you delete the list that you had selected, navigating to the `Lists` page should show the no list selected text.
